modules/iam-role-for-service-accounts-eks/main.tf

```diff
@@ -7,7 +7,7 @@ locals {
   partition           = data.aws_partition.current.partition
   dns_suffix          = data.aws_partition.current.dns_suffix
   region              = data.aws_region.current.name
-  role_name_condition = var.role_name != null ? var.role_name : "${var.role_name_prefix}*"
+  role_name_condition = try(coalesce(var.role_name, "${var.role_name_prefix}*"), null)
```

Comment:
It will never be `null`, because there is always at least `*`, so `try()` is obsolete.

Reply:
I think you are thinking of the `karpenter_controller_cluster_id`/`karpenter_controller_cluster_name` which defaults to `"*"` - `role_name` and `role_name_prefix` both default to `null` which will cause `coalesce()` to fail if neither are provided a value

Reply:
oh, or perhaps you are referring to the asterisk on the prefix variant - this too fails when `role_name_prefix` is `null`, stating that a `null` value cannot be supplied to a templated value

Reply:
Right, then it is well-written. :)